src/core/Akka.Tests/Pattern/CircuitBreakerSpec.cs

```diff
@@ -199,65 +200,86 @@ public class AnAsynchronousCircuitBreakerThatIsClosed : CircuitBreakerSpecBase
 
         [Fact(DisplayName = "An asynchronous circuit breaker that is closed must reset failure count after success")]
         public async Task Must_reset_failure_count_after_success()
         {
             var breaker = MultiFailureCb();
-            _ = breaker.Instance.WithCircuitBreaker(() => Task.Run(SayHi));
-            Enumerable.Range(1, 4).ForEach(_ => breaker.Instance.WithCircuitBreaker(() => Task.Run(ThrowException)));
-            await AwaitAssertAsync(() => breaker.Instance.CurrentFailureCount.ShouldBe(4), AwaitTimeout);
-            _ = breaker.Instance.WithCircuitBreaker(() => Task.Run(SayHi));
-            await AwaitAssertAsync(() => breaker.Instance.CurrentFailureCount.ShouldBe(0), AwaitTimeout);
+            _ = await breaker.Instance.WithCircuitBreaker(SayHiAsync).WaitAsync(AwaitTimeout);
+
+            await Task.WhenAll(Enumerable.Range(1, 4)
+                .Select(_ 
+                    => InterceptException<TestException>(() => breaker.Instance.WithCircuitBreaker(ThrowExceptionAsync))));
+
+            breaker.Instance.CurrentFailureCount.Should().Be(4);
+            
+            var result = await breaker.Instance.WithCircuitBreaker(SayHiAsync).WaitAsync(AwaitTimeout);
+            result.Should().Be("hi");
+            breaker.Instance.CurrentFailureCount.ShouldBe(0);
         }
 
         [Fact(DisplayName = "An asynchronous circuit breaker that is closed must increment failure count on callTimeout")]
         public async Task Must_increment_failure_count_on_callTimeout()
         {
             var breaker = ShortCallTimeoutCb();
 
-            var future = breaker.Instance.WithCircuitBreaker(async () =>
-            {
-                await Task.Delay(150);
-                ThrowException();
-            });
+            var innerFuture = SlowThrowing();
+            var future = breaker.Instance.WithCircuitBreaker(() => innerFuture);
 
-            Assert.True(CheckLatch(breaker.OpenLatch));
-            breaker.Instance.CurrentFailureCount.ShouldBe(1);
+            CheckLatch(breaker.OpenLatch).Should().BeTrue();
+            breaker.Instance.CurrentFailureCount.Should().Be(1);
 
             // Since the timeout should have happened before the inner code finishes
             // we expect a timeout, not TestException
-            await InterceptException<TimeoutException>(() => future.WaitAsync(AwaitTimeout));
+            await InterceptException<TimeoutException>(() => future);
+            
+            // Issue https://github.com/akkadotnet/akka.net/issues/7358
+            // The actual exception is thrown out-of-band with no handler because inner Task is detached
+            // after a timeout and NOT protected
+            
+            // In the bug, the task is still running when it should've been stopped. 
+            innerFuture.IsCompleted.Should().BeTrue();
+            innerFuture.IsFaulted.Should().BeTrue();
+            innerFuture.Exception.Should().BeOfType<TestException>();
```

Comment:
The actual test, the inner Task should have stopped when the circuit breaker timed out, but it is still running as a detached Task.